pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -400,6 +405,17 @@ func (a *Actuator) Update(ctx context.Context, cluster *clusterv1.Cluster, machi
 		return errors.Errorf("failed to ensure tags: %+v", err)
 	}
 
+	scope.MachineStatus.InstanceState = &instanceDescription.State
+
+	if err := a.reconcileLBAttachment(scope, machine, instanceDescription); err != nil {
+		return errors.Errorf("failed to reconcile load balancer attachment: %+v", err)
+	}
+
+	if machine.Spec.ProviderID == nil || *machine.Spec.ProviderID == "" {
```

Comment:
Given that this is repeated, can we do it outside here and the Create call, what do you think about moving it outside?